### DISCUSSION TITLE
[DEPRECATED] default export is deprecated, instead import { shallow } from'zustand/shallow'

### DISCUSSION THREAD

**Discussion by TheBrightKnight** [xyflow/xyflow#2739] [Q&A] [answered]
Hi all, 

I started to work with react-flow today, made some amazing progress and for some reason out of the blue I'm getting hundreds of the following console warning : 

`[DEPRECATED] default export is deprecated, instead import { shallow } from'zustand/shallow'`

I narrowed it down to <ReactFlow />  
I addition at the same time, custom nodeTypes stopped showing the connections between nodes, when moving to default nodeType it shows again. 

It's important to mention again that I did not change anything, to verify I even took the basic tailwind example from the website, created an entire new component and still same issue.  

Any help is appreciated.

**Comment by moklick** [comment, marked as answer]
Hey!

That message comes from Zustand and it should only be there in dev (they released a new version [today](https://github.com/pmndrs/zustand/releases/tag/v4.3.1)). However we will adjust our code and publish a new version with the adjusted import statement. It's only a warning. You can ignore it for now.

Custom nodes types stopped working? Is this related somehow?

**Reply by TheBrightKnight** [reply]
Thanks @moklick ! 
I can only assume it is related though I can’t be sure - I tried everything and can’t get the connection lines between the nodes to show up. Once removing the custom node the lines appear

**Reply by moklick** [reply]
Can you create a codesandbox for this?

**Reply by TheBrightKnight** [reply]
@moklick here it is : 
https://codesandbox.io/s/relaxed-glade-7z930c?file=/src/FlowChart.jsx


I changed some things but it's the same idea; 
Removing the "type : 'custom' " from line 149 brings back the connection lines 

<img width="1166" alt="image" src="https://user-images.githubusercontent.com/17460146/211667712-ebd93aff-2528-4076-afc3-10ba6dd21ec1.png">


<img width="1191" alt="image" src="https://user-images.githubusercontent.com/17460146/211667746-ad84bb3f-96ab-45db-98ec-c4c151d6f408.png">

**Reply by moklick** [reply]
Got it! The issue here is that custom nodes don't have default styles. You need to style them: https://codesandbox.io/s/blissful-proskuriakova-2lxndx?file=/src/styles.css:59-153

**Comment by TheBrightKnight** [comment]
Thanks for the help @moklick; Your codesandbox still doesn't show the connector lines, they are messed up : 

<img width="1204" alt="image" src="https://user-images.githubusercontent.com/17460146/211675525-b0b7ebe1-9f99-4ee1-9262-fd4d3175ec26.png">

While removing the "type: custom" sorts it out 

<img width="1483" alt="image" src="https://user-images.githubusercontent.com/17460146/211675597-4ae9fa68-4e4b-4d4b-8657-bb43870c9861.png">

**Reply by TheBrightKnight** [reply]
Managed to fix it  - https://codesandbox.io/s/eager-torvalds-s6x99o?file=/src/Flow.jsx 

Not sure why, maybe some race condition in the original useEffect I have no idea... 
Thanks again for the help

**Comment by TheBrightKnight** [comment]
This is already been fixed in the latest release;

בתאריך יום ה׳, 19 בינו׳ 2023 ב-11:30 מאת houstonp ***@***.***

**Reply by ghost** [reply]
Cool thank you very much.
